### DISCUSSION TITLE
Referencing images in email using `resourcesPath` doesn't work.

### DISCUSSION THREAD

**Discussion by mallowigi** [keycloak/keycloak#33373] [Q&A]
Hi, I've been struggling for quite a while over a simple problem.

In our company we have developed a Keycloak theme for login and sending emails. In the ftl templates, we sometimes have to display images, so to achieve that we're using the `url.resourcesPath` as stated in the documentation.

This works well for login, but for emails, we're getting a broken image instead. After a lot of research, I found out that the reason is that the `resourcesPath` is returning a path rather than a url, meaning that it would still use the current domain rather than the domain where the keycloak server is deployed.

Now this works great for login and welcome page, as the users usually navigate in these pages. But for emails, it won't work because emails are usually hosted elsewhere. Even in email clients it's simply returned like this: ` <img class="logo-img" src="/resources/2ycvi/email/adm-login/img/logo.png" alt="Logo">`, e.g. lacking the keycloak server domain! 

I've been struggling about this. I first tried to use base64 embedded images, but these don't work in Gmail/Outlook. Then I tried to use an embedded svg, and again no success. But first and foremost I don't understand how something like this is **encouraged by the documentation when it doesn't work**.

I'd be glad to see if other people encountered this issue. My last resort would be to use absolute hardcoded urls but this is clearly a bad solution.

Thanks

**Comment by codespearhead** [comment]
Could you please [close this discussion](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#closing-a-discussion) in favor of this issue [1]? There's already a PR in the works [2].

Make sure to upvote both the issue and the PR by adding a 👍 to their description to help them gain visibility.

[1] https://github.com/keycloak/keycloak/issues/33198
[2] https://github.com/keycloak/keycloak/pull/33366

**Comment by mallowigi** [comment]
Great to hear! I has already resorted to use absolute urls but I'll definitely try this once it's merged
